traitsui/qt4/instance_editor.py

```diff
@@ -365,9 +368,15 @@ def dispose(self):
                     dispatch="ui"
                 )
 
-            self.factory.observe(
-                self.rebuild_items, "values.items", remove=True, dispatch="ui"
-            )
+            # _choice can also be a QLineEdit in which case we never set up
+            # this observer.
+            if isinstance(self._choice, QtGui.QComboBox):
+                self.factory.observe(
+                    self.rebuild_items,
+                    "values.items",
+                    remove=True,
+                    dispatch="ui"
+                )
```

Comment:
Do we also want to remove the other observer hooked up if `factory.name != ""`? Ref https://github.com/enthought/traitsui/blob/6c0e71797a4efa6d74c317c1971e5e61d2944306/traitsui/qt4/instance_editor.py#L109-L112

Reply:
I believe this is already getting unhooked above